auth/services/selfsigned/types.go

```diff
@@ -0,0 +1,133 @@
+/*
+ * Nuts node
+ * Copyright (C) 2023 Nuts community
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ */
+
+package selfsigned
+
+import (
+	"encoding/json"
+	"github.com/nuts-foundation/go-did/vc"
+	"github.com/nuts-foundation/nuts-node/auth/contract"
+	"github.com/nuts-foundation/nuts-node/vcr"
+)
+
+// ContractFormat is the contract format type
+const ContractFormat = "selfsigned"
+
+// VerifiablePresentationType is the dummy verifiable presentation type
+const VerifiablePresentationType = "NutsSelfSignedPresentation"
+
+// SessionCreated represents the session state after creation
+const SessionCreated = "created"
+
+// SessionInProgress represents the session state after rendering the html
+const SessionInProgress = "in-progress"
+
+// SessionCompleted represents the session state after the user has accepted the contract
+const SessionCompleted = "completed"
+
+type SessionStore interface {
+	contract.Signer
+	contract.VPVerifier
+}
+
+// SessionStore is a contract signer and verifier that always succeeds
+// The SessionStore signer is not supposed to be used in a clustered context unless consecutive calls arrive at the same instance
+type sessionStore struct {
+	sessions map[string]session
+	vcr      vcr.VCR
+}
+
+// NewSessionStore returns an initialized SessionStore
+func NewSessionStore(vcr vcr.VCR) SessionStore {
+	return sessionStore{
+		sessions: map[string]session{},
+		vcr:      vcr,
+	}
+}
+
+// session contains the contract text and session signing status
+type session struct {
+	contract string
+	status   string
+	Employer string
+	Employee Employee `json:"employee"`
```

Comment:
wait, we're marshalling "session"? (can't find where). Better to separate that (we can marshal Employee, but take it from the session then marshal it, to avoid accidentally exposing internal session state)

Reply:
Only for unmarshalling. But I splitted the two.